src/tplr/neurons.py

```diff
@@ -209,14 +209,10 @@ async def catchup_with_aggregation_server(
                 logger.warning(
                     f"Failed to process aggregation data for window {current_step}"
                 )
-                # Still advance the optimizer and scheduler
-                instance.optimizer.step()
-                instance.scheduler.step()
+                # Don't advance the optimizer and scheduler
```

Comment:
If we find the aggregated gradient but fail to apply it we could still step here I think.